### PR TITLE
fix: CXSPA-5565(Consent) : If enable all consent when there has one consent, it will prompt wrong message 

### PR DESCRIPTION
**Pull request by i53577**
jira: https://jira.tools.sap/browse/CXSPA-5565

### CODE REVIEW

File: projects/storefrontlib/cms-components/myaccount/consent-management/components/consent-management.component.ts [hunk 1]
```diff
@@ -276,6 +276,14 @@ export class ConsentManagementComponent implements OnInit, OnDestroy {
   allowAll(templates: ConsentTemplate[] = []): void {
     const consentsToGive: ConsentTemplate[] = [];
     templates.forEach((template) => {
+      const givenDate = template.currentConsent?.consentGivenDate;
```
Comment (i53577) [inline projects/storefrontlib/cms-components/myaccount/consent-management/components/consent-management.component.ts:279]: fix in v1

File: projects/storefrontlib/cms-components/myaccount/my-account-v2/my-account-v2-consent-management/components/my-account-v2-consent-management.component.ts [hunk 2]
```diff
@@ -275,10 +275,16 @@ export class MyAccountV2ConsentManagementComponent
   allowAll(templates: ConsentTemplate[] = []): void {
     const consentsToGive: ConsentTemplate[] = [];
     templates.forEach((template) => {
+      const givenDate = template.currentConsent?.consentGivenDate;
```
Comment (i53577) [inline projects/storefrontlib/cms-components/myaccount/my-account-v2/my-account-v2-consent-management/components/my-account-v2-consent-management.component.ts:278]: fix in v2

File: projects/storefrontlib/cms-components/myaccount/my-account-v2/my-account-v2-consent-management/components/my-account-v2-consent-management.component.ts [hunk 3]
```diff
@@ -275,10 +275,16 @@ export class MyAccountV2ConsentManagementComponent
...
+        (givenDate && !withdrawnDate) ||
+        (givenDate && withdrawnDate && givenDate > withdrawnDate);
       if (
-        template.currentConsent &&
-        this.userConsentService.isConsentWithdrawn(template.currentConsent) &&
-        this.isRequiredConsent(template)
+        (template.currentConsent &&
+          this.userConsentService.isConsentWithdrawn(template.currentConsent) &&
+          this.isRequiredConsent(template)) ||
+        isConsentGiven
```
Comment (i53577) [inline projects/storefrontlib/cms-components/myaccount/my-account-v2/my-account-v2-consent-management/components/my-account-v2-consent-management.component.ts:287]: when the consent has been given, do not give it the second time

**Review by niehuayang — APPROVED** [review]
LGTM, Can we keep consistency both v1 and v2?

**Comment by i53577** [comment]
This function is slightly different in v1 and v2. But yes, the bug is fixed both in v1 and v2.

**Pull request merged**